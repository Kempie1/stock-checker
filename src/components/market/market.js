import React, {  useState, useEffect  } from 'react';
import {  useAuth  } from "../../providers/AuthProvider"
import { BrowserRouter as Router, Link} from "react-router-dom"
import axios from 'axios'
import './market.css'
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";
import Loader from "react-loader-spinner";

const Market = (props) => {
const [tickers, setTickers] = useState([])
const { currentUser} = useAuth()
var loading= true

useEffect(()=>{
  axios.get(`https://stockcheckerdb.herokuapp.com/getavailable/`)
  .then(res => {
    setTickers(res.data.Stocks);
    
  })
  .catch(error => {
    if (error.response) {
      // Request made and server responded
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else if (error.request) {
      // The request was made but no response was received
      console.log(error.request);
    } else {
      // Something happened in setting up the request that triggered an Error
      console.log('Error', error.message);
    }
  }
  );

}, [])

function gainOrLoss(open,close){
  loading = false
  if(close-open>0){
    return "loss"
  }
  if(close-open<0){
    return "gain"
  }
  if(close-open===0){
    return "same"
  }
}


//returns the proper icon
function differenceArrow(open,close){
  if(close-open>0){
    return "fa-arrow-down"
  }
  if(close-open<0){
    return "fa-arrow-up"
  }
  if(close-open===0){
    return "fa-equals"
  }
}
//repeated multiple times for extra icons
return (
    <div className="stock_blocks">
    {
      tickers.map((value) => {
        return (
        <Link to={"/stock/"+value[0]}>
          <li key={value[0]} className="stock_block">
            <p>{value[1]}</p>
            <p>({value[0]})</p>
            <p>Day's Volume: {value[2]}</p>
            <p className={gainOrLoss(value[4],value[3])}>Day's change: {(value[4]-value[3]).toFixed(2)} <i className={"fas "+differenceArrow(value[4],value[3])}></i></p>
            </li>
        </Link>
        )
      })
    }
    {
      tickers.map((value) => {
        return (
        <Link to={"/stock/"+value[0]}>
          <li key={value[0]+10} className="stock_block">
            <p>{value[1]}</p>
            <p>({value[0]})</p>
            <p>Day's Volume: {value[2]}</p>
            <p className={gainOrLoss(value[4],value[3])}>Day's change: {(value[4]-value[3]).toFixed(2)} <i className={"fas "+differenceArrow(value[4],value[3])}></i></p>
            </li>
        </Link>
        )
      })
    }
    {
      tickers.map((value) => {
        return (
        <Link to={"/stock/"+value[0]}>
          <li key={value[0]+20} className="stock_block">
            <p>{value[1]}</p>
            <p>({value[0]})</p>
            <p>Day's Volume: {value[2]}</p>
            <p className={gainOrLoss(value[4],value[3])}>Day's change: {(value[4]-value[3]).toFixed(2)} <i className={"fas "+differenceArrow(value[4],value[3])}></i></p>
            </li>
        </Link>
        )
      })
    }
    <Loader
        className="loader"
        visible={loading}
        type="Circles"
        color="#063D74"
        height={100}
        width={100}
      />
    </div>
    
  );
};

export default Market;