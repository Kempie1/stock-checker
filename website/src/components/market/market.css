.stock_block{
    margin-top: 1rem;
    margin-left: 1rem;
    display: inline-block;
    width: 15rem;
    height:10rem;
    padding: 1rem;
    border-style: solid;
    box-shadow: 10px 10px 5px grey;
}

  .loss{ color:#EC748C; }
  .gain{ color:#68B691; }
  