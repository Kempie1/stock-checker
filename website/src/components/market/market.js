import React, {  useState, useEffect  } from 'react';
import {  useAuth  } from "../../providers/AuthProvider"
import { BrowserRouter as Router, Link, useHistory } from "react-router-dom"
import axios from 'axios'

const Market = (props) => {
const [tickers, setTickers] = useState("")
const { currentUser} = useAuth()

useEffect(()=>{
  axios.get(`https://stockcheckerdb.herokuapp.com/getavailable/`)
  .then(res => {
    setTickers(res.data.Stocks);
  })
  .catch(error => {
    if (error.response) {
      // Request made and server responded
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else if (error.request) {
      // The request was made but no response was received
      console.log(error.request);
    } else {
      // Something happened in setting up the request that triggered an Error
      console.log('Error', error.message);
    }
  }
  );
}, [])
//[Array(1), Array(1), Array(1)]
function listStocks(){
  //limit 10
  //var count = 0
  if (Array.isArray(tickers)){
  tickers.map((value) => {
    console.log(value[0])
    return <p>{value}</p>}
  )
  }
  }



return (
    <div>
    <Link to="/stock/AAPL">
      Apple
    </Link>
    {listStocks()}
    </div>
    
  );
};

export default Market;