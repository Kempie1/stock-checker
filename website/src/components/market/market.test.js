import React from "react"
import { render } from '@testing-library/react';
import {Market, gainOrLoss, differenceArrow, getAvailable, makeBlocks} from "./market"
import { BrowserRouter as Router, Link} from "react-router-dom"
import TestRenderer from 'react-test-renderer';

const axios = require('axios')
jest.mock('axios')
const data = require('../../mocks/available.json');



test("gainOrLoss", ()=>{
    expect(gainOrLoss(10,3)).toBe("loss")
    expect(gainOrLoss(3,10)).toBe("gain")
    expect(gainOrLoss(10,10)).toBe("same")
})

test("differenceArrow", ()=>{
    expect(differenceArrow(10,3)).toBe("fa-arrow-down")
    expect(differenceArrow(3,10)).toBe("fa-arrow-up")
    expect(differenceArrow(10,10)).toBe("fa-equals")
})

test("makeBlocks matches snapshot", ()=>{
    const mockblock=["A", "Agilent Technologies, Inc.", "264724", "132.31", "131.3"]
    const component = render (
    <Router>
        {makeBlocks(mockblock)}
    </Router>
    )
    expect(component.container).toMatchSnapshot();
})

test("<Market/> doesn't crash", ()=>{
    const div = document.createElement("div");
    render (<Market test={true} tickers={data} />,div)
})


test ("mock api call getavailable", ()=>{
    //arange
    const mockedResponse = data
    axios.get.mockResolvedValue(mockedResponse)
    const marke = require ('./market')
    //act
    marke.getAvailable()
    //asserts
    expect(axios.get).toHaveBeenCalled()
    expect(axios.get).toHaveBeenCalledWith(`https://stockcheckerdb.herokuapp.com/getavailable/`)

})

test("<Market/> matches snapshot", ()=>{
    const renderedComponent = TestRenderer.create(<Market test={true}  tickers={data} />)
    expect(renderedComponent).toMatchSnapshot()

})

