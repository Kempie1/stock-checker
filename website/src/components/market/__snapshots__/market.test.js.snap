// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Market/> matches snapshot 1`] = `
<div
  className="stock_blocks"
/>
`;

exports[`makeBlocks matches snapshot 1`] = `
<div>
  <a
    href="/stock/A"
    id="A"
  >
    <li
      class="stock_block"
    >
      <p>
        Agilent Technologies, Inc.
      </p>
      <p>
        ( 
        A
         )
      </p>
      <p>
        Day's Volume: 
        264724
      </p>
      <p
        class="loss"
      >
        Day's change: 
        -1.01
        <i
          class="fas fa-arrow-down"
        />
      </p>
    </li>
  </a>
</div>
`;
